src/main/java/org/truffleruby/parser/lexer/RubyLexer.java

```diff
@@ -3089,7 +3089,7 @@ public void tokAdd(int first_byte, RopeBuilder buffer) {
     }
 
     public void tokCopy(int length, RopeBuilder buffer) {
-        buffer.append(parserRopeOperations.makeShared(lexb, lex_p - length, length));
+        buffer.append(ArrayUtils.extractRange(lexb.getBytes(), lex_p - length, lex_p));
```

Comment:
Does `lexb` always have it's `byte[] bytes` field computed?

Reply:
Yes. We end up having to get the bytes to tokenize across the board. But any substring of a multi-byte string also forces `byte[]` materialization so the substring's code range can be determined.